clients/deltalake/spark3/pom.xml

```diff
@@ -118,9 +118,37 @@
   </repositories>
 
   <build>
-    <sourceDirectory>${project.build.directory}/sources</sourceDirectory>
-    <testSourceDirectory>${project.build.directory}/test-sources</testSourceDirectory>
     <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
```

Comment:
What's the advantage of this over the previous approach?

Reply:
The previous approach only looked in the generated-sources. This looks in both `main/src` and the generated ones.

Reply:
got it